articles/quickstart/native/android/03-session-handling.md

```diff
@@ -140,30 +152,64 @@ The `SecureCredentialsManager` can prompt the user for local device authenticati
 
 ## Log the User Out
 
-To log the user out, you remove their credentials and navigate them to the login screen. When using a Credentials Manager you do that calling `clearCredentials`.
+To log the user out, it is normally enough to remove their credentials and navigate them back to the login screen. When using a Credentials Manager you do that calling `clearCredentials`. In addition, you could ask the `WebAuthProvider` to remove the cookie set by the Browser at authentication time, so that the users are forced to re-enter their credentials the next time they try to authenticate. The sample combines these two strategies.
 
-In the sample, the LoginActivity checks that a boolean extra is present in the Intent at the Activity launch, so that if this flag is true the credentials are first removed from the Credentials Manager.
+Check in the LoginActivity if a boolean extra is present in the Intent at the Activity launch. This scenario triggered by the MainActivity dictates that the user wants to log out.
 
 ```java
 // app/src/main/java/com/auth0/samples/MainActivity.java
 
 private void logout() {
     Intent intent = new Intent(this, LoginActivity.class);
-    intent.putExtra(LoginActivity.KEY_CLEAR_CREDENTIALS, true);
+    intent.putExtra(LoginActivity.EXTRA_CLEAR_CREDENTIALS, true);
     startActivity(intent);
     finish();
 }
 
 // app/src/main/java/com/auth0/samples/LoginActivity.java
+
 @Override
 protected void onCreate(Bundle savedInstanceState) {
     //...
-    if (getIntent().getBooleanExtra(KEY_CLEAR_CREDENTIALS, false)) {
-        credentialsManager.clearCredentials();
+
+    if (getIntent().getBooleanExtra(EXTRA_CLEAR_CREDENTIALS, false)) {
+        doLogout();
+        return;
     }
 }
+
+private void doLogout() {
+    WebAuthProvider.logout(auth0)
+        .withScheme("demo")
+        .start(this, logoutCallback);
+}
+
+private VoidCallback logoutCallback = new VoidCallback() {
+    @Override
+    public void onSuccess(Void payload) {
+        credentialsManager.clearCredentials();
+    }
+
+    @Override
+    public void onFailure(Auth0Exception error) {
+        //Log out canceled, keep the user logged in
+        showNextActivity();
+    }
+};
+
 ```
 
+
+The logout is achieved by using the `WebAuthProvider` class. This call will open the Browser and navigate the user to the logout endpoint. If the log out is cancelled, you might want to take the user back to where they were before attempting to log out. If the call succeeded you will remove the credentials from the manager instance.
+
+
 ::: note
 If you are not using our Credentials Manager classes, you are responsible for ensuring that the user's credentials have been removed.
 :::
+
+
+As you did previously for the login step, you will need to whitelist as well this logout URL in the dashboard.
+
+<%= include('../../../_includes/_logout_url', { returnTo: 'demo://${account.namespace}/android/YOUR_APP_PACKAGE_NAME/callback' }) %>
```

Comment:
I suspect this is the line failing the build. Can you confirm @davidpatrick if it is OK to use `${ ... }` inside a string and inside `<% .. %>` please?

Reply:
Maybe:

```
`demo://${account.namespace}/android/YOUR_APP_PACKAGE_NAME/callback`
```

Tickmarks?